### PR TITLE
add check and test for minimum asset amounts for new option types

### PR DESCRIPTION
**Pull request by Flip-Liquid**

### CODE REVIEW

**Comment by 0xAlcibiades** [comment]
@Flip-Liquid rather than a min asset amount, we should just do a min fee of 1 wei when the fee results in a 0 calculation.

File: src/interfaces/IOptionSettlementEngine.sol [hunk 1]
```diff
@@ -200,6 +200,13 @@ interface IOptionSettlementEngine {
      */
     error InvalidAssets(address asset1, address asset2);
 
+    /**
+     * @notice The supplied underlying and exercise amounts are too small.
+     * @param underlyingAmount The amount of the underlying asset.
+     * @param exerciseAmount The amount of the exercise asset.
+     */
+    error InvalidAssetAmounts(uint256 underlyingAmount, uint256 exerciseAmount);
```
Comment (neodaoist) [inline src/interfaces/IOptionSettlementEngine.sol:208]: Nit — slight pref for `InvalidAmounts`, better symmetry with `InvalidAssets` imo but I can see either way.

Comment (Flip-Liquid) [reply src/interfaces/IOptionSettlementEngine.sol:208]: n/a, going to be removing the error type

File: src/interfaces/IOptionSettlementEngine.sol [hunk 2]
```diff
@@ -200,6 +200,13 @@ interface IOptionSettlementEngine {
      */
     error InvalidAssets(address asset1, address asset2);
 
+    /**
```
Comment (0xAlcibiades) [inline src/interfaces/IOptionSettlementEngine.sol:203]: @Flip-Liquid this can go away.

**Review by 0xAlcibiades — CHANGES_REQUESTED** [review]

**Review by 0xAlcibiades — APPROVED** [review]
